src/rgw/rgw_acl_swift.cc

```
@@ -59,56 +59,133 @@ static bool uid_is_public(string& uid)
+//      auto firstpart = uid.substr(0, pos);	/* project or .r: */
+//      auto secondpart = uid.substr(pos + 1);	/* user or... */
+      if (uid[0] != '.') {
+	rgw_user user(uid);
+        std::string empty_string;
+	grant.set_canon(user, empty_string, perm);
+      } else {
+	/* might be a HTTP referrer-based acl.  Not now (ever?) */
+// need to check firstpart to see if it matches .referrer cases.
+//	bool is_negative = false;
+//	if (*secondpart == '-') {
+//	  secondpart = secondpart.substr(1);
+//	  is_negative = true;
+//	}
+//	grant.set_referrer(secondpart, perm);
+        ldout(cct, 10) << "Unknown designator: " << uid << dendl;
+	if (!result) result = -EUNKNOWN_DESIGNATOR;
+      }
+      acl.add_grant(&grant);
+    } else {
       rgw_user user(uid);
       if (rgw_get_user_info_by_uid(store, user, grant_user) < 0) {
+	// also catches ".rlistings" case - no separate mech for this in ceph
         ldout(cct, 10) << "grant user does not exist:" << uid << dendl;
-        /* skipping silently */
+        /* skipping not so silently */
+	if (!result) result = -ENO_SUCH_USER;
       } else {
         grant.set_canon(user, grant_user.display_name, perm);
         acl.add_grant(&grant);
       }
     }
   }
+  return result;
 }
 
-bool RGWAccessControlPolicy_SWIFT::create(RGWRados *store, rgw_user& id, string& name, string& read_list, string& write_list)
+int RGWAccessControlPolicy_SWIFT::create(RGWRados *store, rgw_user& id, string& name, const char* read_list, const char* write_list, uint32_t& rw_mask)
 {
   acl.create_default(id, name);
   owner.set_id(id);
   owner.set_name(name);
+  rw_mask = 0;
+  int result = 0;
 
-  if (read_list.size()) {
+  if (read_list) {
     list<string> uids;
     int r = parse_list(read_list, uids);
     if (r < 0) {
-      ldout(cct, 0) << "ERROR: parse_list returned r=" << r << dendl;
-      return false;
+      ldout(cct, 0) << "ERROR: parse_list returned r=" << r << " rl=" << read_list << dendl;
+      result = r;
```

why not return these failures immediately?